packages/math-extras/src/index.ts

```diff
@@ -0,0 +1,174 @@
+import './pointExtras';
+import './rectangleExtras';
+
+import { IPointData, Point } from '@pixi/math';
+
+/**
+ * The idea of a relative epsilon comparison is to find the difference between the two numbers,
+ * and see if it is less than `Math.EPSILON`.
+ *
+ * _Note: Only available with **@pixi/math-extras**._
+ *
+ * @param {number} a - First floating number to compare.
+ * @param {number} b - Second floating number to compare.
+ * @returns {boolean} Returns `true` if the difference between the values is less than `Math.EPSILON`; otherwise `false`.
+ */
+export function floatEqual(a: number, b: number): boolean;
+/**
+ * The idea of a relative epsilon comparison is to find the difference between the two numbers,
+ * and see if it is less than a given epsilon.
+ * A good epsilon would be the N% of the largest of the two values or `Math.EPSILON`.
+ *
+ * _Note: Only available with **@pixi/math-extras**._
+ *
+ * @memberof PIXI
+ * @param {number} a - First floating number to compare.
+ * @param {number} b - Second floating number to compare.
+ * @param {number} epsilon - The epsilon to compare to.
+ * The larger the epsilon, the easier for the numbers to be considered equals.
+ * @returns {boolean} Returns `true` if the difference between the values is less than the given epsilon;
+ * otherwise `false`.
+ */
+export function floatEqual(a: number, b: number, epsilon: number): boolean;
+export function floatEqual(a: number, b: number, epsilon: number = Number.EPSILON): boolean
+{
+    if (a === b)
+    {
+        return true;
+    }
+
+    const diff = Math.abs(a - b);
+
+    return diff < epsilon;
+}
+
+/**
+ * Generic line or segment intersection.
+ * A line can intersect outside the two points defining it, the segment can't.
+ * @param aStart - First point of the first line.
+ * @param aEnd - Second point of the first line.
+ * @param bStart - First point of the second line.
+ * @param bEnd - Second point of the second line.
+ * @param isLine - Set to true if you want Line (unbounded) intersection.
+ * @param {IPointData} [outPoint] - A Point-like object in which to store the value,
+ * optional (otherwise will create a new Point).
+ * @returns {IPointData} The point where the lines/segments intersect or a `NaN` Point.
+ */
+function genericLineIntersection<T extends IPointData>(
+    aStart: IPointData,
+    aEnd: IPointData,
+    bStart: IPointData,
+    bEnd: IPointData,
+    isLine: boolean,
+    outPoint?: T): T
+{
+    if (!outPoint)
+    {
+        (outPoint as any) = new Point();
+    }
+
+    const denominator = (((bEnd.y - bStart.y) * (aEnd.x - aStart.x)) - ((bEnd.x - bStart.x) * (aEnd.y - aStart.y)));
+
+    // If lines are parallel or overlapping, the intersection can be nowhere or everywhere... NaN.
+    if (denominator === 0)
```

Comment:
We could, however `settings.EPSILON` isn't a thing. I used `Math.EPSILON` for the`floatEqual` in `packages/math-extras/src/index.ts`
Since that function is from this same package, I think it would be good to use it 🤔

Reply:
My bad, that's sounds good.